ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql;
+
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import io.confluent.ksql.util.QueryMetadata;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The context in which statements can be executed.
+ */
+public interface KsqlExecutionContext {
```

Comment:
What do you think about adding the service context type as a type parameter here. Then `KsqlEngine` would implement `KsqlExecutionContext<DefaultServiceContext>`, and `SandboxedExecutionContext` would implement `KsqlExecutionContext<SandboxedExecutionContext>`. This lets us do compile-time verification that we're using the right contexts in the right places. So `RequestValidator`'s constructor could then have the signature:

```
private RequestValidator(
        final KsqlExecutionContext<SandboxedExecutionContext> executionSandbox,
        final ServiceContext serviceContext,
        final KsqlConfig ksqlConfig,
        final Map<String, Object> propertyOverrides
    );
```
And we're then guaranteed that we never ever validate with an actual engine.

Reply:
Humm... I'm not sure. I'll have a think about this. I'm not convinced at first pass, as I'm not sure it gains up much.  The request validator is only created in one place, (admittedly only at the moment).  I'm not sure the type of service context should be exposed as a generic param.  The same benefit could be achieved by just having `RequestValidator` take a `SandboxedExecutionContext`.